src/faultlog/faultlog_main.cpp

```diff
@@ -41,17 +49,146 @@ int main(int /*arg*/, char** /*argv*/)
+            int manualGuardCount =
+                GuardWithoutEidRecords::getCount(unresolvedRecords);
+            int unresolvedPelsCount = UnresolvedPELs::getCount(bus);
+            int deconfigCount = DeconfigRecords::getCount();
+
+            if ((guardCount > 0) || (manualGuardCount > 0) ||
+                (unresolvedPelsCount > 0) || (deconfigCount > 0))
+            {
+                std::unordered_map<std::string, std::string> data;
+                data.emplace("GUARD_WITH_ASSOC_ERROR_COUNT",
+                             std::to_string(guardCount));
+                data.emplace("GUARD_WITH_NO_ASSOC_ERROR_COUNT",
+                             std::to_string(manualGuardCount));
+                data.emplace("UNRESOLVED_PEL_WITH_DECONFIG_BIT_COUNT",
+                             std::to_string(unresolvedPelsCount));
+                data.emplace("DECONFIG_RECORD_COUNT",
+                             std::to_string(deconfigCount));
+
+                auto method = bus.new_method_call(
+                    "xyz.openbmc_project.Logging",
+                    "/xyz/openbmc_project/logging",
+                    "xyz.openbmc_project.Logging.Create", "Create");
+                method.append("org.open_power.Faultlog.Error.DeconfiguredHW",
+                              Severity::Error, data);
+                auto reply = method.call();
+                if (reply.is_method_error())
+                {
+                    lg2::error("Error in calling D-Bus method to create PEL");
+                }
+                lg2::info("faultlog {GUARD_COUNT}, {MAN_GUARD_COUNT}, "
+                          "{DECONFIG_COUNT} , {PEL_COUNT} ",
+                          "GUARD_COUNT", guardCount, "MAN_GUARD_COUNT",
+                          manualGuardCount, "DECONFIG_COUNT", deconfigCount,
+                          "PEL_COUNT", unresolvedPelsCount);
+            }
+            else
+            {
+                lg2::info("There are no pending service actions ignoring "
+                          "creating fautlog pel");
+            }
+        }
+        // write faultlog json to stdout
+        else if (listFaultlog)
+        {
+            (void)GuardWithEidRecords::populate(bus, unresolvedRecords,
+                                                faultLogJson);
+            (void)GuardWithoutEidRecords::populate(unresolvedRecords,
+                                                   faultLogJson);
+
+            (void)FaultLogPolicy::populate(bus, faultLogJson);
+
+            (void)UnresolvedPELs::populate(bus, unresolvedRecords,
+                                           faultLogJson);
+            (void)DeconfigRecords::populate(faultLogJson);
+        }
+        else
+        {
+            lg2::error("Invalid option");
+        }
+
+        if (listFaultlog || deconfig || unresolvedPels || policy ||
+            guardWithoutEid || guardWithEid)
+        {
+            std::cout << faultLogJson.dump(2) << std::endl;
```

Comment:
not to trace?

Reply:
dump plugin will collect this output to json file so need to do stdout